wren-ai-service/src/web/v1/services/ask.py

```diff
@@ -215,11 +248,23 @@ async def ask(
                         configurations=ask_request.configurations,
                     )
 
-                valid_generation_results = []
                 if sql_valid_results := text_to_sql_generation_results["post_process"][
                     "valid_generation_results"
                 ]:
-                    valid_generation_results += sql_valid_results
+                    valid_sql_summary_results = (
+                        await self._add_summary_to_sql_candidates(
+                            sql_valid_results,
+                            ask_request.query,
+                            ask_request.configurations.language,
+                        )
+                    )
+                    api_results += [
+                        AskResult(**result) for result in valid_sql_summary_results
+                    ]
+                    self._ask_results[query_id] = AskResultResponse(
+                        status="generating",
+                        response=api_results,
+                    )
```

Comment:
i think the results might be more than three. Do we need to remove any after the third candidate?